htdocs/public/payment/newpayment.php

```diff
@@ -2423,49 +2422,28 @@
 			  }
 			};
 
-			var cardElement = elements.create('card', {style: style});
+			var paymentElement = elements.create("payment");
 
 			// Add an instance of the card Element into the `card-element` <div>
-			cardElement.mount('#card-element');
-
-			// Handle real-time validation errors from the card Element.
-			cardElement.addEventListener('change', function(event) {
-				var displayError = document.getElementById('card-errors');
-				  if (event.error) {
-					  console.log("Show event error (like 'Incorrect card number', ...)");
-					displayError.textContent = event.error.message;
-				  } else {
-					  console.log("Reset error message");
-					displayError.textContent = '';
-				  }
-			});
+			paymentElement.mount("#payment-element");
 
 			// Handle form submission
-			var cardholderName = document.getElementById('cardholder-name');
 			var cardButton = document.getElementById('buttontopay');
-			var clientSecret = cardButton.dataset.secret;
 
 			cardButton.addEventListener('click', function(event) {
 				console.log("We click on buttontopay");
 				event.preventDefault();
 
-				if (cardholderName.value == '')
-				{
-					console.log("Field Card holder is empty");
-					var displayError = document.getElementById('card-errors');
-					displayError.textContent = '<?php print dol_escape_js($langs->trans("ErrorFieldRequired", $langs->transnoentitiesnoconv("CardOwner"))); ?>';
-				}
-				else
-				{
 					/* Disable button to pay and show hourglass cursor */
 					jQuery('#hourglasstopay').show();
 					jQuery('#buttontopay').hide();
 
-					stripe.handleCardPayment(
-					clientSecret, cardElement, {
+					stripe.confirmPayment({
+		  elements,confirmParams: {
+		  return_url: '<?php echo $urlok; ?>',
 						payment_method_data: {
 							billing_details: {
-								name: cardholderName.value
+								name: 'test'
```

Comment:
This looks strange. Does this means we do not propagate the name of credit card owner anymore ?

Reply:
@eldy nope! The New element  form don’t use it